### ISSUE TITLE
Summarize memory blocks and total consumption

### ISSUE THREAD

**Issue by PizzaProgram** [winsiderss/systeminformer#1841]
### Description of the feature, modification, idea or suggestion

I'm currently trying to track a memory leak in my app, and System Informer is a great help to do that.
Except that I have to calculate 1000+ lines of memory block consumption one by one, when I select them.
It would be great to:
## Summarize the selected lines at "Memory" page and show at the bottom.

![kép](https://github.com/winsiderss/systeminformer/assets/26026089/26370a9a-9d5b-4ff2-8f14-0c144de0d5ce)

Even more important would be, to
## Show total memory consumption at Status page

 - Total size 
 - Total delta
 - Peak total

![kép](https://github.com/winsiderss/systeminformer/assets/26026089/12d2f423-ef4b-4d99-b4a4-2f7bba0975f3)
Because private memory allocation alone is not enough to measure, if the app is reaching a total of 2GB and crashes.

Maybe consider to add a checkbox Graph to the "Performance" page too, to:
 - Show total memory consumption (private + virtual)

Optional:
 - Memory fragmentation statistics

**Comment by PizzaProgram** [comment]
It would be great, if the "Total bytes" memory consumption would be calculated the same way as [VMMap](https://learn.microsoft.com/en-us/sysinternals/downloads/vmmap) does from Sysinternals.

**Comment by dmex** [comment]
>  the "Total bytes" memory consumption

"Total bytes" are under the I/O category for disk/network and unrelated to memory usage?

**Comment by PizzaProgram** [comment]
> "Total bytes" are under the I/O category for disk/network and unrelated to memory usage?

Sorry, but I don't understand this question.
I'm talking only about memory consumption. 
VMMap is a memory-usage monitoring app too.
I've never mentioned anything about I/O nor disk/network.

I've added this line:
> It would be great, if the "Total bytes" memory consumption would be calculated the same way as [VMMap](https://learn.microsoft.com/en-us/sysinternals/downloads/vmmap) does from Sysinternals.

only for the reason to my original request, because maybe "simply summarizing private bytes + virtual bytes" is not the correct way to calculate "total memory consumption" ...

**Comment by dmex** [comment]
> It would be great, if the "Total bytes" memory consumption would be calculated the same way as [VMMap](https://learn.microsoft.com/en-us/sysinternals/downloads/vmmap)

"Total bytes" is the name of disk/network statistics 🤷‍♂️ 

![image](https://github.com/winsiderss/systeminformer/assets/1306177/63aa28e1-cdb9-4d8d-96cf-9070540733c5)

> "Total bytes" memory consumption would be calculated the same way as [VMMap](https://learn.microsoft.com/en-us/sysinternals/downloads/vmmap) does from Sysinternals.

The values down the left column?

![image](https://github.com/winsiderss/systeminformer/assets/1306177/5ddc332e-c702-4097-8c87-d3feb0dc37e8)

or the values across the top row?

![image](https://github.com/winsiderss/systeminformer/assets/1306177/a6521edd-de72-4737-aa8e-03324e1c00f8)

>  maybe "simply summarizing private bytes + virtual bytes" is not the correct way to calculate "total memory consumption"

The total would be `Private bytes + Private WS + Shareable WS + Shared WS`?

![image](https://github.com/winsiderss/systeminformer/assets/1306177/3c7f8530-00b8-41d4-a83c-69eb9c853d51)

![image](https://github.com/winsiderss/systeminformer/assets/1306177/929dd3ac-c73a-4aef-b2d9-33befd4b2cb7)

VMMap and System Informer are querying the system at different intervals and the program or the system changed a small number of pages between those intervals but otherwise the totals the same?

**Comment by PizzaProgram** [comment]
> "Total bytes" is the name of disk/network statistics 🤷‍♂️

Total size memory = Total bytes memory
It does not really matter, how we call it. It can be the very same "naming", the only thing that matters, where is it listed.
 _(Whether under "I/O" or under "Memory")
The different size/byte naming is currently confusing.

IMHO to be consistent everything should be listed like this:_
- Private (bytes)
- Private delta (bytes)
- Total (bytes)
- ... _etc_
- Page faults (count)
- ... _etc_

So the unit could be separated by a sign. Also "Private [bytes]" is ok.
Than we wouldn't need to thing about if "byte" or "size" is better.

**Comment by PizzaProgram** [comment]
> The values down the left column?

YES, but I don't think it would be necessary to list **all** those values.
Also maybe it would consume too much CPU to collect those separately.
 _(Sadly I know very little about memory consumption procedures to know how they are counted, which data can be easily obtained from OS, etc...)_
I can reflect only as "end user" to the fact that:  "Total memory consumption" is missing from the list.

> The total would be Private bytes + Private WS + Shareable WS + Shared WS?

I'm not sure. Can we ask the programmers of VMMap?
If I count together those columns, they do not match. 
 (For example the "Shareable" blue line: `12700+2216 != 24884`)

![kép](https://github.com/winsiderss/systeminformer/assets/26026089/26cc2778-40c1-4c9c-a9ba-49b98728d356)

**Comment by PizzaProgram** [comment]
Answer v2.0:
> The total would be `Private bytes + Private WS + Shareable WS + Shared WS`?

Sorry, If you have ment by that the System Informer's total, than than the answer is: No
If I summarize 109+67+25+11 = 212MB
 .. even if I would add `Working set` +92 = 304MB, 
is still much smaller than **349MB real total** consumption calculated by VMMap.

So something is missing... But I don't know what.

![kép](https://github.com/winsiderss/systeminformer/assets/26026089/4e5549dd-fb33-471c-8897-8d040f3e31d3)

**Comment by jxy-s** [comment]
VMMap `Size` column includes blocks of reserved but uncommitted memory. That isn't displayed in the `Statistics` tab in System Informer. But if you can see reserved memory in the `Memory` tab:

![image](https://github.com/winsiderss/systeminformer/assets/11687482/dc38c40e-8e28-4b1b-9c43-e3a90d28af89)

![image](https://github.com/winsiderss/systeminformer/assets/11687482/470f0bc7-6fed-42ac-a471-175eb557409d)
